File: backend/app/schemas/product_import.py
from __future__ import annotations

from decimal import Decimal

from pydantic import BaseModel, Field


class ProductImportRow(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    price: Decimal = Field(ge=0)
    category: str | None = Field(default=None, max_length=120)
    manufacturer: str | None = Field(default=None, max_length=120)
    size: str | None = Field(default=None, max_length=120)


class ProductImportParseResponse(BaseModel):
    columns: list[str] = Field(default_factory=list)
    mapping: dict[str, str | None] = Field(default_factory=dict)
    rows: list[ProductImportRow] = Field(default_factory=list)


class ProductImportConfirmRequest(BaseModel):
    rows: list[ProductImportRow] = Field(min_length=1)


class ProductImportError(BaseModel):
    row_index: int
    name: str
    message: str


class ProductImportConfirmResponse(BaseModel):
    created: int
    errors: list[ProductImportError] = Field(default_factory=list)

File: frontend/types/product-import.ts
export type ProductImportRow = {
  name: string;
  price: string | number;
  category?: string | null;
  manufacturer?: string | null;
  size?: string | null;
};

export type ProductImportParseResponse = {
  columns: string[];
  mapping: Record<string, string | null>;
  rows: ProductImportRow[];
};

export type ProductImportConfirmResponse = {
  created: number;
  errors: Array<{
    row_index: number;
    name: string;
    message: string;
  }>;
};

export type ImportSource = "excel" | "pdf" | "photo";

export const IMPORT_FIELD_LABELS: Record<string, string> = {
  name: "Название товара",
  price: "Цена продажи",
  category: "Категория",
  manufacturer: "Производитель",
  size: "Размер",
};

export const IMPORT_SOURCE_LABELS: Record<ImportSource, string> = {
  excel: "Excel / CSV",
  pdf: "PDF",
  photo: "Фото",
};

export const IMPORT_ACCEPT: Record<ImportSource, string> = {
  excel: ".xlsx,.csv",
  pdf: ".pdf",
  photo: ".png,.jpg,.jpeg,.webp",
};
